pkgs/os-specific/linux/dcgm/default.nix

```diff
@@ -0,0 +1,108 @@
+{ lib
+, callPackage
+, gcc11Stdenv
+, fetchFromGitHub
+, addOpenGLRunpath
+, catch2
+, cmake
+, cudaPackages_10_2
+, cudaPackages_11_8
+, cudaPackages_12
+, fmt_9
+, git
+, jsoncpp
+, libevent
+, plog
+, python3
+, tclap_1_4
+, yaml-cpp
+}:
+let
+  # Flags copied from DCGM's libevent build script
+  libevent-nossl = libevent.override { sslSupport = false; };
+  libevent-nossl-static = libevent-nossl.overrideAttrs (super: {
+    CFLAGS = "-Wno-cast-function-type -Wno-implicit-fallthrough -fPIC";
+    CXXFLAGS = "-Wno-cast-function-type -Wno-implicit-fallthrough -fPIC";
+    configureFlags = super.configureFlags ++ [ "--disable-shared" "--with-pic" ];
+  });
+
+  jsoncpp-static = jsoncpp.override { enableStatic = true; };
```

Comment:
why are these build flags necessary? if at all possible, it is preferable to just use dependencies as they are default packaged

Reply:
DCGM is *very* particular about its build. Its entire build system is actually deterministic, but it uses Docker to achieve that. This makes it extremely particular about the exact build flags used for every dependency. These dependencies are built with these specific flags and I ran into build failures in other configurations. While it might be theoretically possible to find a different configuration that succeeds, this one exactly matches what upstream does in its build system so it is the most likely to succeed.